src/Service/PageListGetter.php

```diff
@@ -135,6 +156,12 @@ protected function runQuery( $params, $contName, $resName, $pageIdName = 'pageid
 
 			// Add the results to the output page list.
 			foreach ( $result['query'][$resName] as $member ) {
+				// Assign negative pageid if page is non-existent.
+				if ( array_key_exists( 'missing', $member ) ) {
```

Comment:
Is `missing` used for this purpose for all of the API calls in this class? I'm wondering if it wouldn't be more universal to check for the absence or emptiness of `$member[$pageIdName]`?